tutorials/content-changes/src/main/java/ContentListening.java

```diff
@@ -0,0 +1,90 @@
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import com.teamdev.jxbrowser.chromium.Browser;
+import com.teamdev.jxbrowser.chromium.JSObject;
+import com.teamdev.jxbrowser.chromium.events.LoadAdapter;
+import com.teamdev.jxbrowser.chromium.events.LoadEvent;
+import com.teamdev.jxbrowser.chromium.swing.BrowserView;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * This example shows how to listen to DOM changes from a Java object.
+ */
+public class ContentListening {
+
+    public static void main(String[] args) {
+        final Browser browser = new Browser();
+        BrowserView view = new BrowserView(browser);
+
+        JFrame frame = new JFrame();
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        frame.add(view, BorderLayout.CENTER);
+        frame.setSize(700, 500);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+
+        browser.addLoadListener(new LoadAdapter() {
+            @Override
+            public void onDocumentLoadedInMainFrame(LoadEvent event) {
+                Browser browser = event.getBrowser();
+                JSObject window = browser.executeJavaScriptAndReturnValue("window")
+                                         .asObject();
+                window.setProperty("java", new JavaObject());
+                String javaScript = load("observer.js");
+                browser.executeJavaScript(javaScript);
+            }
+        });
+
+        String html = load("index.html");
+        browser.loadHTML(html);
+    }
+
+    /**
+     * The object observing DOM changes.
+     */
+    private static class JavaObject {
+
+        @SuppressWarnings("unused") // invoked by callback processing code.
+        void onDomChanged() {
+            System.out.println("DOM node changed");
+        }
+    }
+
+    /**
+     * Loads a resource content as a string.
+     */
+    private static String load(String resourceFile) {
+        URL url = Resources.getResource(resourceFile);
+        String result;
+        try {
+            result = Resources.toString(url, Charsets.UTF_8);
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to load resource " + resourceFile, e);
+        }
+        return result;
+    }
+}
```

Comment:
Replace the code with the following:
```java
/*
 *  Copyright 2018, TeamDev. All rights reserved.
 *
 *  Redistribution and use in source and/or binary forms, with or without
 *  modification, must retain the above copyright notice and the following
 *  disclaimer.
 *
 *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
 *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
 *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
 *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
 *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

import com.google.common.base.Charsets;
import com.google.common.io.Resources;
import com.teamdev.jxbrowser.chromium.Browser;
import com.teamdev.jxbrowser.chromium.JSObject;
import com.teamdev.jxbrowser.chromium.events.LoadAdapter;
import com.teamdev.jxbrowser.chromium.events.LoadEvent;
import com.teamdev.jxbrowser.chromium.swing.BrowserView;

import javax.swing.*;
import java.awt.*;
import java.io.IOException;
import java.net.URL;

/**
 * This example shows how to listen to DOM changes from a Java object.
 */
public class ContentListening {

    public static void main(String[] args) {
        final Browser browser = new Browser();
        BrowserView view = new BrowserView(browser);

        JFrame frame = new JFrame();
        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        frame.add(view, BorderLayout.CENTER);
        frame.setSize(700, 500);
        frame.setLocationRelativeTo(null);
        frame.setVisible(true);

        browser.addScriptContextListener(new ScriptContextAdapter() {
            @Override
            public void onScriptContextCreated(ScriptContextEvent event) {
                Browser browser = event.getBrowser();
                JSObject window = browser.executeJavaScriptAndReturnValue("window")
                                         .asObject();
                window.setProperty("java", new JavaObject());
                String javaScript = load("observer.js");
                browser.executeJavaScript(javaScript);
            }
        });

        String html = load("index.html");
        browser.loadHTML(html);
    }

    /**
     * The object observing DOM changes.
     */
    public static class JavaObject {

        @SuppressWarnings("unused") // invoked by callback processing code.
        public void onDomChanged(String innerHTML) {
            System.out.println("DOM node has been changed");
        }
    }

    /**
     * Loads a resource content as a string.
     */
    private static String load(String resourceFile) {
        URL url = Resources.getResource(resourceFile);
        String result;
        try {
            result = Resources.toString(url, Charsets.UTF_8);
        } catch (IOException e) {
            throw new IllegalStateException("Unable to load resource " + resourceFile, e);
        }
        return result;
    }
}
```